src/main/java/org/fcrepo/transform/transformations/LDPathTransform.java

```diff
@@ -78,60 +76,57 @@ public LDPathTransform(final InputStream query) {
         this.query = query;
     }
 
-    /**
-     * Pull a node-type specific transform out of JCR
-     * @param node the node
-     * @param key the key
-     * @return node-type specific transform
-     * @throws RepositoryException if repository exception occurred
-     */
-    public static LDPathTransform getNodeTypeTransform(final Node node,
-        final String key) throws RepositoryException {
-
-        final Node programNode = node.getSession().getNode(CONFIGURATION_FOLDER + key);
+    public static LDPathTransform getResourceTransform(final FedoraResource resource, final NodeService nodeService,
+            final String key) throws RepositoryException {
 
-        LOGGER.debug("Found program node: {}", programNode.getPath());
+        final FedoraResource transformResource =
+                nodeService.find(resource.getNode().getSession(), CONFIGURATION_FOLDER + key);
 
-        final NodeType primaryNodeType = node.getPrimaryNodeType();
+        LOGGER.debug("Found transform resource: {}", transformResource.getPath());
 
-        final Set<NodeType> supertypes = orderedBy(BY_NAME).add(primaryNodeType.getSupertypes()).build();
-        final Set<NodeType> mixinTypes = orderedBy(BY_NAME).add(node.getMixinNodeTypes()).build();
+        final List<URI> rdfTypes = resource.getTypes();
 
-        // start with mixins, primary type, and supertypes of primary type
-        final ImmutableList.Builder<NodeType> nodeTypesB = builder();
-        nodeTypesB.addAll(mixinTypes).add(primaryNodeType).addAll(supertypes);
+        LOGGER.debug("Discovered rdf types: {}", rdfTypes);
 
-        // add supertypes of mixins
-        mixinTypes.stream().map(mixin -> orderedBy(BY_NAME).add(mixin.getDeclaredSupertypes()).build())
-            .forEach(nodeTypesB::addAll);
+        final NamespaceRegistry nsRegistry = resource.getNode().getSession().getWorkspace().getNamespaceRegistry();
 
-        final List<NodeType> nodeTypes = nodeTypesB.build();
-
-        LOGGER.debug("Discovered node types: {}", nodeTypes);
-        for (final NodeType nodeType : nodeTypes) {
-            if (programNode.hasNode(nodeType.toString())) {
-                return new LDPathTransform(programNode.getNode(nodeType.toString())
-                                               .getNode(JCR_CONTENT)
-                                               .getProperty(JCR_DATA)
-                                               .getBinary().getStream());
+        // convert rdf:type with URI namespace to prefixed namespace
+        final Function<URI, String> namespaceUriToPrefix = x -> {
+            final String uriString = x.toString();
+            try {
+                for (final String namespace : Arrays.asList(nsRegistry.getURIs())) {
+                    // There is an empty namespace URI and that matches everything.
```

Comment:
Ok, I played with this and the blank namespace does match.
With the `namespace.length() > 0` left in and some debug statements outside and inside the loop I see:
```
DEBUG 08:16:29.474 (LDPathTransform) namespace is 
DEBUG 08:16:29.474 (LDPathTransform) namespace is http://www.w3.org/1999/02/22-rdf-syntax-ns#
DEBUG 08:16:29.474 (LDPathTransform) namespace is http://www.jcp.org/jcr/mix/1.0
DEBUG 08:16:29.474 (LDPathTransform) namespace is http://www.jcp.org/jcr/1.0
DEBUG 08:16:29.474 (LDPathTransform) namespace is http://www.w3.org/ns/ldp#
DEBUG 08:16:29.474 (LDPathTransform) namespace is http://www.w3.org/2001/XMLSchema
DEBUG 08:16:29.474 (LDPathTransform) namespace is http://www.modeshape.org/images/1.0
DEBUG 08:16:29.475 (LDPathTransform) namespace is http://www.w3.org/2000/01/rdf-schema#
DEBUG 08:16:29.475 (LDPathTransform) namespace is http://purl.org/dc/elements/1.1/
DEBUG 08:16:29.475 (LDPathTransform) namespace is http://www.w3.org/2000/xmlns/
DEBUG 08:16:29.475 (LDPathTransform) namespace is http://www.ebu.ch/metadata/ontologies/ebucore/ebucore#
DEBUG 08:16:29.475 (LDPathTransform) namespace is http://www.loc.gov/premis/rdf/v1#
DEBUG 08:16:29.475 (LDPathTransform) namespace is http://www.jcp.org/jcr/sv/1.0
DEBUG 08:16:29.475 (LDPathTransform) namespace is info:fedora/test/
DEBUG 08:16:29.475 (LDPathTransform) namespace is http://fedora.info/definitions/v4/repository#
DEBUG 08:16:29.475 (LDPathTransform) namespace match on http://fedora.info/definitions/v4/repository# to URI http://fedora.info/definitions/v4/repository#Container
```

If I remove the `namespace.length() > 0` I end up with:
```
DEBUG 08:17:28.101 (LDPathTransform) namespace is ()
DEBUG 08:17:28.101 (LDPathTransform) namespace match on () to URI (http://fedora.info/definitions/v4/repository#Container)
```

Perhaps I shouldn't be using `.startsWith()`?